### PR TITLE
disable auto-unidenting of closing curly braces

### LINKED ISSUE #99: wrong indentation of closing brace
E.g. try typing the following:
```
f1:{
    something
    };
```
The closing curly brace will be auto-unindented, even though this results in code that won't load, worse still, you won't notice this immediately as this doesn't break over IPC.

### PR DESCRIPTION
**Pull request by gyorokpeter**
Fixes #99 

THE FOLLOWING DISCLAIMER APPLIES TO ALL SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE:
THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.
ONLY THE SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE, IF ANY, THAT ARE ATTACHED TO (OR OTHERWISE ACCOMPANY) THIS SUBMISSION (AND ORDINARY COURSE CONTRIBUTIONS OF FUTURES PATCHES THERETO) ARE TO BE CONSIDERED A CONTRIBUTION. NO OTHER SOFTWARE CODE OR MATERIALS ARE A CONTRIBUTION.

### CODE REVIEW

File: src/main/studio/qeditor/RSTokenMaker.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public Token getTokenList(Segment text, int initialTokenType, int startOffset) {
 
     @Override
     public boolean getCurlyBracesDenoteCodeBlocks(int languageIndex) {
-        return true;
+        return false;   //this needs to be false, otherwise the closing brace of functions will jump to the beginning of the line, which results in a syntax error
```
Comment (dzmipt) [inline src/main/studio/qeditor/RSTokenMaker.java:66]: I thought may be we can somehow override the algorithm of adding curling braces to add at least one space if comes on the beginning of the line?

File: src/main/studio/qeditor/RSTokenMaker.java [hunk 2]
```diff
@@ -14,6 +14,7 @@ public class RSTokenMaker extends TokenMakerBase {
     public static final String CONTENT_TYPE = "text/q";
 
     private QSyntaxParser parser = new QSyntaxParser();
+    public static boolean unindentCurlyBraces = false;
```
Comment (dzmipt) [inline src/main/studio/qeditor/RSTokenMaker.java:17]: I didn't find a code which would initialize the variable from the settings on start-up. It looks this variable is updated only if I call Settings Dialog.  

Also generally it is not encourage in the Java world to have public access to variables. Usually they are doing via setter functions...

Comment (gyorokpeter) [reply src/main/studio/qeditor/RSTokenMaker.java]: Updated, can you check again?

**Review by dzmipt — APPROVED** [review]

File: src/main/studio/qeditor/RSTokenMaker.java (hunk 2 above)
Comment (dzmipt) [reply src/main/studio/qeditor/RSTokenMaker.java]: I think that's good now.

**Pull request merged**